### PR TITLE
[TS] Improve TextInput TS

### PR DESCRIPTION
**Pull request by fredhase**
#### What does this PR do?
Improve the types at the src/js/components/TextInput/index.d.ts

#### Where should the reviewer start?
the file: src/js/components/TextInput/index.d.ts

#### What testing has been done on this PR?
Test that already exist and manual test

#### How should this be manually tested?
Testing the TextInput in the storybook

#### Any background context you want to provide?
Yes. I couldn't find the type to 'suggestion' manually testing it. I've tryed some things here and there, but none was successful. Also, everytime it was showing as 'undefined'. But maybe, probably(?), I did something not quite right.

#### What are the relevant issues?
#3165 

#### Do the grommet docs need to be updated?
I believe there's no need.

#### Should this PR be mentioned in the release notes?
I believe there's no need.

#### Is this change backwards compatible or is it a breaking change?
I believe it is.

### CODE REVIEW

File: src/js/components/TextInput/index.d.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ export interface TextInputProps {
   messages?: {enterSelect?: string,suggestionsCount?: string,suggestionsExist?: string,suggestionIsOpen?: string};
   name?: string;
   onSelect?: ((x: { target: React.RefObject<HTMLElement>['current'], suggestion: any }) => void);
-  onSuggestionsOpen?: ((...args: any[]) => any);
-  onSuggestionsClose?: ((...args: any[]) => any);
+  onSuggestionsOpen?: (() => Function);
+  onSuggestionsClose?: (() => Function);
```
Comment (ShimiSun) [inline src/js/components/TextInput/index.d.ts]: This suggested type wouldn't work on the code example of https://storybook.grommet.io/?path=/story/textinput--custom

Comment (negreirosleo) [reply src/js/components/TextInput/index.d.ts]: @ShimiSun why wouldnt it work? I tested this here and it worked the same way

Comment (fredhase) [reply src/js/components/TextInput/index.d.ts]: Sorry, I didn't understand. I tested it manually and it seemed to work the same way it does in the link, or in the master branch :thinking: 
Could you explain, please? :)

Comment (ShimiSun) [reply src/js/components/TextInput/index.d.ts]: Are you sure? this is what I'm getting when adding Custom.tsx file with the content of https://storybook.grommet.io/?path=/story/textinput--custom and using the suggested fix
image
```
Type 'void' is not assignable to type 'Function'.ts(2322)
```

Comment (fredhase) [reply src/js/components/TextInput/index.d.ts]: Oh. Now I see it. Just made a commit with the changes, everything seems fine now :)
Thanks again

Comment (ShimiSun) [reply src/js/components/TextInput/index.d.ts]: Can you please add a tsx story file that can validate this change? similarly to what I've shown above.

Comment (fredhase) [reply src/js/components/TextInput/index.d.ts]: Sure. I submmited the commit before I saw your message on the other PR. I'll add in both

Comment (fredhase) [reply src/js/components/TextInput/index.d.ts]: The file is now added :)

File: src/js/components/TextInput/stories/typescript/Custom.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+          />
+        </Box>
+      </Box>
+    </Grommet>
+  );
+};
+
+storiesOf('TextInput', module).add('Custom', () => (
+  <CustomSuggestionsTextInput />
+));
```
Comment (ShimiSun) [inline src/js/components/TextInput/stories/typescript/Custom.tsx]: Can you please take a look at how other tsx files are taking care of publishing to storybook?
They include the isChromatic query and place the story under the TypeScript directory.

File: src/js/components/TextInput/index.d.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ export interface TextInputProps {
   messages?: {enterSelect?: string,suggestionsCount?: string,suggestionsExist?: string,suggestionIsOpen?: string};
   name?: string;
   onSelect?: ((x: { target: React.RefObject<HTMLElement>['current'], suggestion: any }) => void);
-  onSuggestionsOpen?: ((...args: any[]) => any);
-  onSuggestionsClose?: ((...args: any[]) => any);
+  onSuggestionsOpen?: (() => void);
+  onSuggestionsClose?: (() => void);
```
Comment (ShimiSun) [inline src/js/components/TextInput/index.d.ts]: @oorestisime I need another pair of eyes to make sure this change is clear.

Comment (oorestisime) [reply src/js/components/TextInput/index.d.ts]: looks good to me. we call it without any args and we don't treat any response so all good here

File: src/js/components/TextInput/stories/typescript/Custom.tsx (hunk 2 above)
Comment (ShimiSun) [reply src/js/components/TextInput/stories/typescript/Custom.tsx]: I pushed a fix for it.

**Comment by ShimiSun** [comment]
Thank you for your contribution!

**Pull request merged**